public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -161,7 +164,10 @@ export class DashboardModel {
 
     this.addBuiltInAnnotationQuery();
     this.sortPanelsByGridPos();
-    this.hasChangesThatAffectsAllPanels = false;
+    this.appEventsSubscription = appEvents.subscribe(VariableChanged, (event) => {
+      this.processRepeats(event.payload.panelIds);
+      this.startRefresh(event.payload.panelIds);
+    });
```

Comment:
Trying a new pattern here, instead of calling functions on the dashboard model the variable system publishes appEvents that the dashboard model can listen to.